ntbs-service/DataAccess/UserRepository.cs

```diff
@@ -63,9 +63,20 @@ private async Task UpdateUser(User existingUser, User newUser, IEnumerable<TBSer
         private static void AddCaseManagerTbServices(User user, IEnumerable<TBService> tbServices)
         {
             var caseManagerTbServices = tbServices
-                .Select(tb => new CaseManagerTbService {TbService = tb, CaseManager = user})
+                .Select(tb => new CaseManagerTbService
+                {
+                    TbService = tb,
+                    CaseManager = user,
+                    TbServiceCode = tb.Code,
+                    CaseManagerUsername = user.Username
+                })
                 .ToList();
-            user.CaseManagerTbServices = caseManagerTbServices.Any() ? caseManagerTbServices : null;
+            if (caseManagerTbServices.Count != user.CaseManagerTbServices.Count ||
+                !caseManagerTbServices.All(c =>
+                    user.CaseManagerTbServices.Any(t => t.Equals(c))))
+            {
+                user.CaseManagerTbServices = caseManagerTbServices.Any() ? caseManagerTbServices : null;

```

Comment:
Think we could go the step further to add/remove individual records needing adding/removing - but very happy to be ignored. I do think this is a good enough compromise - for something which should be relatively static.

Reply:
This now has a fix up if you want to take another look